airflow/executors/base_executor.py

```diff
@@ -54,6 +57,44 @@
 # Task tuple to send to be executed
 TaskTuple = Tuple[TaskInstanceKey, CommandType, Optional[str], Optional[Any]]
 
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class RunningRetryAttemptType:
+    """
+    For keeping track of attempts to queue again when task still apparently running.
+
+    We don't want to slow down the loop, so we don't block, but we allow it to be
+    re-checked for at least MIN_SECONDS seconds.
+    """
+
+    MIN_SECONDS = 5
```

Comment:
Is this long enough?

Reply:
who knows!? what do you suggest... 30?

Reply:
i made it 10.

i think that should be more than enough. but we can always change it in future if nec.